pkg/resources/usercluster/deployment.go

```diff
@@ -131,6 +131,7 @@ func DeploymentCreator(data userclusterControllerData, openshift bool) reconcili
 				"-version", data.Cluster().Spec.Version.String(),
 				"-cloud-provider-name", data.GetKubernetesCloudProviderName(),
 				"-owner-email", data.Cluster().Status.UserEmail,
+				fmt.Sprintf("-enable-ssh-key-agent=%t", data.Cluster().Spec.EnableUserSSHKeyAgent),
```

Comment:
Does this work for nil values?

Reply:
Well, if the field in the API Cluster is nil, then this field will be defaulted to true:
https://github.com/kubermatic/kubermatic/pull/6426/files#diff-27c41fe06286392f13c14c7d60150fdb798847490bc8122cad718e44b810a589R181-R186